sunpy/map/mapbase.py

```diff
@@ -847,6 +847,8 @@ def resample(self, dimensions, method='linear'):
             new_meta['CD2_2'] *= scale_factor_y
         new_meta['crpix1'] = (dimensions[0].value + 1) / 2.
         new_meta['crpix2'] = (dimensions[1].value + 1) / 2.
+        new_meta['naxis1'] = dimensions[1].value
```

Comment:
My point is that this test should not need these keys. There is no need for us to add the `naxis` keys to the header, only update them if they are already there.